hive-llap/llap.sh

```diff
@@ -0,0 +1,497 @@
+    LLAP_XMX_6_INT=${LLAP_XMX_6%.*}
+
+    ### jvm headroom for the llap executors
+    if  (( $LLAP_XMX_6_INT > 6144 )); then
+    LLAP_HEADROOM=6114
+        else
+    LLAP_HEADROOM=$LLAP_XMX_6_INT
+    fi
+    echo "LLAP daemon headroom: ${LLAP_HEADROOM}"
+
+    ##cache is whatever is left over after heardroom and executor memory is accounted for
+    LLAP_CACHE=$(expr ${LLAP_SIZE} - ${LLAP_HEADROOM} - ${LLAP_XMX})
+
+
+    ##if there is no additional room, then no cache will be used
+    if (( $LLAP_CACHE < 0 )); then
+    LLAP_CACHE=0
+    fi
+    echo "LLAP in-memory cache: ${LLAP_CACHE}"
+
+    
+        ###if user didn't pass in num llap instances, take worker node count -1
+        if [[ -z $NUM_LLAP_NODES ]]; then
+            LLAP_INSTANCES=$(expr ${WORKER_NODE_COUNT} - 1) 
+        else
+            LLAP_INSTANCES=$NUM_LLAP_NODES
+        fi 
+
+    echo "LLAP daemon instances: ${LLAP_INSTANCES}"
+
+    echo "Starting LLAP..."
+    sudo -u hive hive --service llap \
+    --instances "${LLAP_INSTANCES}" \
+    --size "${LLAP_SIZE}"m \
+    --executors "${LLAP_EXECUTORS}" \
+    --xmx "${LLAP_XMX}"m \
+    --cache "${LLAP_CACHE}"m \
+    --name llap0 \
+    --auxhbase=false \
+    --skiphadoopversion \
+    --directory /tmp/llap_staging \
+    --output /tmp/llap_output \
+    --loglevel INFO \
+    --startImmediately 
+    fi
+}
+
+##main driver function for the script
+function configure_llap(){
+
+if [[ "${HOSTNAME}" == "${LLAP_MASTER_FQDN}" ]]; then
+    echo "running primary master config...."
+    pre_flight_checks
+    download_init_actions
+    package_tez_lib_uris
+    add_yarn_service_dir
+    configure_yarn_site
+    configure_core_site
+    configure_hive_site
+    configure_SSD_caching_master
+    replace_core_llap_files
+    get_log4j
+    configure_tez_site_xml
+    return 0
+fi
+
+if [[ "${ROLE}" == "Worker" ]]; then
+    echo "running worker config...."
+    pre_flight_checks
+    configure_yarn_site
+    configure_core_site
+    configure_hive_site
+    configure_SSD_caching_worker
+    get_log4j
+    configure_tez_site_xml
+    replace_core_llap_files
+    return 0
+fi
+
+if [[ "${ROLE}" == "Master" && "${HOSTNAME}" != "${LLAP_MASTER_FQDN}"  ]]; then
+    echo "running master config...."
+    pre_flight_checks
+    configure_yarn_site
+    configure_core_site
+    configure_hive_site
+    get_log4j
+    configure_tez_site_xml
+    replace_core_llap_files
+    return 0
+fi
+
+}
+
+###run llapstatus command to determine if running
+function wait_for_llap_ready() {
+    if [[ "${HOSTNAME}" == "${LLAP_MASTER_FQDN}" ]]; then
+        echo "wait for LLAP to launch...."
+        sudo -u hive hive --service llapstatus --name llap0 -w -r 1 -i 5
+        echo "LLAP started...."
+    else
+        echo "skipping...."
+    fi
+}
+
+echo "Running configuration process...."
+configure_llap
+
+echo "Starting LLAP...."
+start_llap
+
+echo "Verify full start...."
+wait_for_llap_ready
```

Comment:
Looks like the same logic is repeated in both llap.sh and llap_restart.sh. Would be better to rename the restart script to `start_llap.sh` and trigger that after configure_llap is done.

Reply:
done